Source/WebCore/html/CanvasBase.cpp

```diff
@@ -358,9 +358,14 @@ RefPtr<ImageBuffer> CanvasBase::allocateImageBuffer(bool usesDisplayListDrawing,
     return ImageBuffer::create(size(), RenderingPurpose::Canvas, 1, colorSpace, pixelFormat, bufferOptions, context);
 }
 
-bool CanvasBase::postProcessPixelBuffer(Ref<PixelBuffer> pixelBuffer, bool wasLastDrawByBitMap, const HashSet<uint32_t>& suppliedColors) const
+bool CanvasBase::shouldInjectNoiseBeforeReadback() const
 {
-    if (!scriptExecutionContext() || !scriptExecutionContext()->noiseInjectionHashSalt() || wasLastDrawByBitMap || !scriptExecutionContext()->settingsValues().canvasNoiseInjectionEnabled)
+    return scriptExecutionContext() && scriptExecutionContext()->noiseInjectionHashSalt();
+}
+
+bool CanvasBase::postProcessPixelBuffer(Ref<PixelBuffer>&& pixelBuffer, bool wasLastDrawByBitMap, const HashSet<uint32_t>& suppliedColors) const
```

Comment:
I'm guessing we don't need a `WTFMove(*pixelBuffer)` in `CanvasRenderingContext2DBase::postProcessPixelBuffer` for this rvalue? It's a good chance, I just wouldn't expect that causes an automatic move.

Reply:
So I think we can't use `WTFMove(*pixelBuffer)` in there, since we use the `pixelBuffer` later. The `&&` was mostly just for consistency (since we generally prefer taking `Ref&&` and `RefPtr&&` as arguments instead of just `Ref` or `RefPtr`, to prevent unnecessary ref-churn in some cases).

Reply:
Oh, right, good point.